### PR TITLE
Conventional Commits

### LINKED ISSUE #219: Use semantic-release
This will afford us a couple of niceties
* Automatic version updates
* Automated changelog generation

## To Do
* [ ] Configure `semantic-release` to use lerna in the project root on merge to master
* [ ] Configure `changelog` publishing on merge to master
* [ ] Create custom prepare/release plugin
  * [ ] Run `lerna-publish`during prepare step
  * [ ] Push to npm during release step for each package

### PR DESCRIPTION
**Pull request by aabenoja**
# Use Conventional Commits

Lerna provides a `--conventional-commits` flag in its api. This allows us to automate version bumping based on commit messages similar to `semantic-release`.  Similar to wtw-im/babel-preset-wtw-im#7 commit messages should follow the [eslint convention](Lerna provides a `--conventional-commits` flag in its api. This allows us to automate version bumping based on commit messages similar to `semantic-release`.  Similar to wtw-im/babel-preset-wtw-im#7) to determine the resulting version.

## Process Summary
- Commit messages are vaildated against `commitlint` - a commit message linter to ensure messages follow the designated structure.
- Master builds will run a new `Publish` job once tests pass
  - `Publish` will first publish the updated packages to npm
  - Once npm packages are updated the styleguide will be published to `gh-pages` with the correct version number

## Concerns
- ~~conventional commits are not very friendly towards micro commits. We may need to require that PRs be squashed down to a single, detailed commit in order to intelligently generate a [bloated changelog](https://gist.github.com/aabenoja/973575c8f4b3e61902c8762504d34d9a).~~ This may not be an issue as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#doesn-t-this-discourage-rapid-development-and-fast-iteration), but it may still make sense to ask for squashes whenever possible. Quoting the link below because firewalls.

> ### Doesn’t this discourage rapid development and fast iteration?
>
> It discourages moving fast in a disorganized way. It helps you be able to move fast long term across multiple projects with varied contributors.

- It is still possible to bypass `commitlint` and wind up with pull requests that don't follow the commit structure. The version bot is useless after this gets merged in, and I'm not sure it's worth much to update the bot to verify at least one commit in a PR follows the convention.

Closes #219

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
   "scripts": {
-    "ci": "sh build.sh",
+    "build": "sh build.sh",
+    "ci": "sh build.sh && sh ci.sh",
     "version_only": "lerna publish --skip-npm --force-publish",
     "publish_only": "lerna publish --skip-git",
     "publish": "lerna publish",
     "link": "lerna link --force-local",
-    "precommit": "lint-staged"
+    "start": "npm start --prefix packages/es-components"
```
Comment (jrios) [inline package.json]: nice. 👍

**Comment by jrios** [comment]
How much should we enforce this moving forward? Should all PRs that are currently submitted update their commit messages? or do we roll forward and require all new PRs to follow it?

**Comment by aabenoja** [comment]
Personally the sooner we get this out the better. A lot of this relies on git tags to determine how far back `conventional-commits` will look to determine the version bump. I think both of @Darrken's PRs are small enough that we can easily get his commits up to date with the new conventions, and I think your own would be simple enough to reword and (maybe) squash.

**Comment by jrios** [comment]
Sounds good. I have no problem rewording my commits. I'd prefer not to squash them though.

**Comment by Darrken** [comment]
Do we want to pull this in before the other PRs then? I can get my commit messages updated.

**Comment by jrios** [comment]
@Darrken when you get a chance, please update your PRs. Thanks!

**Comment by Darrken** [comment]
Already done!

**Pull request merged**